knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -1,51 +1,240 @@
-	log.Info("Deleting ConsoleCLIDownload CR for kn")
-	manifest, err := mfc.NewManifest(manifestPath(), apiclient, mf.UseLogger(log.WithName("mf")))
+// applyKnConsoleCLIDownload applies kn ConsoleCLIDownload by finding
+// kn download resource route URL and populating spec accordingly
+func applyKnConsoleCLIDownload(apiclient client.Client, namespace string) error {
+	route := &routev1.Route{}
+	knRoute := ""
+
+	err := wait.PollImmediate(2*time.Second, 20*time.Second, func() (bool, error) {
+		err := apiclient.Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: knConsoleCLIDownloadDeployRoute}, route)
+		switch {
+		case apierrors.IsNotFound(err):
+			return false, nil
+		case err != nil:
+			return false, err
+		default:
+			knRoute = getCanonicalHost(route)
+			if knRoute == "" {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+
 	if err != nil {
-		return fmt.Errorf("failed to read ConsoleCLIDownload manifest: %w", err)
+		return fmt.Errorf("failed to find kn ConsoleCLIDownload deployment route")
 	}
-	if err := manifest.Delete(); err != nil {
-		return fmt.Errorf("failed to delete ConsoleCLIDownload manifest: %w", err)
+
+	log.Info("Creating kn ConsoleCLIDownload CR")
+	knConsoleObj := populateKnConsoleCLIDownload(https(knRoute))
+	err = apiclient.Create(context.TODO(), knConsoleObj)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create kn ConsoleCLIDownload CR: %w", err)
 	}
 	return nil
 }
 
+// Delete deletes kn ConsoleCLIDownload CO and respective deployment resources
+func Delete(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) error {
+	log.Info("Deleting kn ConsoleCLIDownload CO")
+	if err := apiclient.Delete(context.TODO(), populateKnConsoleCLIDownload("")); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to delete kn ConsoleCLIDownload CO: %w", err)
+	}
+
+	log.Info("Deleting kn ConsoleCLIDownload resources")
+	manifest, err := manifest(instance, apiclient, scheme)
+	if err != nil {
+		return err
+	}
+
+	if err := manifest.Delete(); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to delete kn ConsoleCLIDownload resources manifest: %w", err)
+	}
+
+	return nil
+}
+
+// manifest returns kn ConsoleCLIDownload deploymnet resources manifest after traformation
+func manifest(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) (mf.Manifest, error) {
+	manifest, err := rawManifest(apiclient)
+	if err != nil {
+		return mf.Manifest{}, fmt.Errorf("failed to read kn ConsoleCLIDownload deployment manifest: %w", err)
+	}
+
+	// 1. Use instance's namespace to deploy download resources into
+	// 2. Set proper kn-cli-artifacts image
+	transforms := []mf.Transformer{mf.InjectNamespace(instance.GetNamespace()),
+		replaceKnCLIArtifactsImage(os.Getenv("IMAGE_KN_CLI_ARTIFACTS"), scheme),
+	}
```

Comment:
should we add here another transformation which adds annotations for its owner OR ownerReferences ?

Reply:
Added annotations for owner as done for kourier.